src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -158,14 +170,63 @@ private static string IssuerValidator(string issuer, SecurityToken securityToken
             return issuer;
         }
 
-        private static bool AudienceValidator(IEnumerable<string> audiences, SecurityToken securityToken, TokenValidationParameters validationParameters)
+        internal static bool AudienceValidator(IEnumerable<string> audiences, SecurityToken securityToken, TokenValidationParameters validationParameters)
         {
             foreach (string audience in audiences)
             {
+                // First see if we have an exact match.
                 if (validationParameters.ValidAudiences.Any(p => string.Equals(audience, p, StringComparison.OrdinalIgnoreCase)))
                 {
                     return true;
                 }
+
+                // In slots scenarios, the hostname of the incoming audience may include a runtime
+                // site name slot component (e.g. https://test__5bb5.azurewebsites.net). We normalize
+                // this by removing the slot component (e.g. "__5bb5") and check again.
+                if (TryGetNormalizedAudience(audience, out string normalizedAudience) &&
+                    validationParameters.ValidAudiences.Any(p => string.Equals(normalizedAudience, p, StringComparison.OrdinalIgnoreCase)))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool TryGetNormalizedAudience(string audience, out string normalizedAudience)
+        {
+            normalizedAudience = null;
+
+            if (!string.IsNullOrEmpty(audience))
```

Comment:
Could we instead use the runtime site name provided by the environment? Does that environment variable provide what we need to get the proper app name? If so, that would be a more reliable way to get this information without needing to parse it from the original audience URI.

Reply:
Can we at least, to be safer, look for the last index of the separator?